redis/cache/redisx/redisxtest/redisxtest.go

```diff
@@ -0,0 +1,64 @@
+package redisxtest
+
+import (
+	"context"
+
+	"github.com/alicebob/miniredis/v2"
+	"github.com/joomcode/redispipe/redis"
+	"github.com/joomcode/redispipe/redisconn"
+
+	"github.com/reddit/baseplate.go/redis/cache/redisx"
+)
+
+// MockRedisCluster wraps a local version of redis
+type MockRedisCluster struct {
+	redisCluster *miniredis.Miniredis
+}
+
+func NewMockRedisCluster() (mockRedisCluster MockRedisCluster, err error) {
+	redisCluster, err := miniredis.Run()
+	if err != nil {
+		return MockRedisCluster{}, err
+	}
+
+	return MockRedisCluster{
+		redisCluster: redisCluster,
+	}, nil
+}
+
+// Addr returns address of mock redis cluster e.g. '127.0.0.1:12345'.
+func (mrc *MockRedisCluster) Addr() string {
+	return mrc.redisCluster.Addr()
+}
+
+// Close shuts down the MockRedisCluster
+func (mrc *MockRedisCluster) Close() error {
+	mrc.redisCluster.Close()
+	return nil
+}
+
+// NewMockRedisClient sets up a client and sender to a mock redis cluster
+func NewMockRedisClient(
+	ctx context.Context,
+	address string,
+	opts redisconn.Opts,
+) (client redisx.BaseSync, teardown func(), err error) {
```

Comment:
I would still recommend change this function signature into:
```
func NewMockRedisClient(ctx context.Context, tb testing.TB, address string, opts redisconn.Opts) redisx.BaseSync
```

it's OK for the mocked cluster to be a global singleton that's initialized in `TestMain` and shared between tests, because a mocked cluster is probably using much more resource and stand it up and tear it down in every test case is probably too slow/wasteful, but for clients, the pattern we should promote should be that every unit test create it and clean it up, instead of doing it in `TestMain`.